code/client/iconimporter

```diff
@@ -86,41 +86,47 @@ def generate_pngs_from_installer_pkg(install_item, repo_path):
                         break
     elif munkicommon.hasValidPackageExt(item_path):
         pkg_path = item_path
+        pkg_repo = repo
     if pkg_path:
-        if os.path.isdir(pkg_path):
-            icon_paths = iconutils.extractAppIconsFromBundlePkg(pkg_path)
+        if (pkg_repo and pkg_repo.isdir(pkg_path)) or (not repo and os.path.isdir(pkg_path)):
+            icon_paths = iconutils.extractAppIconsFromBundlePkg(pkg_path, pkg_repo)
         else:
+            handle = None
+            if pkg_repo:
+                handle = pkg_repo.open(pkg_path, 'r')
+                pkg_path = handle.local_path
             icon_paths = iconutils.extractAppIconsFromFlatPkg(pkg_path)
 
     if mountpoint:
         munkicommon.unmountdmg(mountpoint)
 
     if len(icon_paths) == 1:
-        png_path = os.path.join(
-            repo_path, u'icons', install_item['name'] + u'.png')
-        result = iconutils.convertIconToPNG(icon_paths[0], png_path)
+        png_path = repo.join(u'icons', install_item['name'] + u'.png')
+        handle = repo.open(png_path, 'w')
+        result = iconutils.convertIconToPNG(icon_paths[0], handle.local_path)
         if result:
             print_utf8(u'\tWrote: %s' % png_path)
     elif len(icon_paths) > 1:
         index = 1
         for icon_path in icon_paths:
-            png_path = os.path.join(
-                repo_path, u'icons',
-                install_item['name'] + '_' + str(index) + u'.png')
-            result = iconutils.convertIconToPNG(icon_path, png_path)
+            png_path = repo.join(
+                u'icons', install_item['name'] + '_' + str(index) + u'.png')
+            handle = repo.open(png_path, 'w')
+            result = iconutils.convertIconToPNG(icon_path, handle.local_path)
             if result:
                 print_utf8(u'\tWrote: %s' % png_path)
             index += 1
     else:
         print_utf8(u'\tNo application icons found.')
 
 
-def find_items_to_check(repo_path, itemlist=None):
+def findItemsToCheck(repo, itemlist=None):
```

Comment:
This function used to be named "find_items_to_check". The new name causes a problem later in the code...